shell/platform/linux/fl_view.cc

```diff
@@ -135,9 +137,12 @@ static void fl_view_constructed(GObject* object) {
   FlView* self = FL_VIEW(object);
 
   GdkDisplay* display = gtk_widget_get_display(GTK_WIDGET(self));
+#ifdef GDK_WINDOWING_X11
   if (GDK_IS_X11_DISPLAY(display)) {
     self->renderer = FL_RENDERER(fl_renderer_x11_new());
-  } else if (GDK_IS_WAYLAND_DISPLAY(display)) {
+  } else
+#endif
+      if (GDK_IS_WAYLAND_DISPLAY(display)) {
```

Comment:
The indenting is weird here - did the formatter change it to be like this? You could just drop the else above if it's confusing things.

Reply:
Yeah, that's the result of `clang-format`. My short experience in that domain is that it's better to agree with it. See another weird example here: https://github.com/flutter/engine/pull/21217#discussion_r489522397

Reply:
OK, please use a helper function to make this clearer, i.e.

```c
static FlRenderer *get_renderer_for_display(GdkDisplay) {
#ifdef GDK_WINDOWING_X11
  if (GDK_IS_X11_DISPLAY(display)) {
    return FL_RENDERER(fl_renderer_x11_new());
  }
#endif

  if (GDK_IS_WAYLAND_DISPLAY(display)) {
    return FL_RENDERER(fl_renderer_wayland_new());
  }

  return nullptr;
}
```

Reply:
I added extra `fl_view_` prefix to `get_renderer_for_display()` as this seems to be the local convention used in this file.